src/js/component.js

```diff
@@ -684,8 +684,8 @@ class Component {
    * @param  {Event|Object|String} event  A string (the type) or an event object with a type attribute
    * @return {Component}       self
    */
-  trigger(event) {
-    Events.trigger(this.el_, event);
+  trigger(event, ...data) {
```

Comment:
I think we're thinking of requiring that data be an options object and just having the one extra parameter for data.